lib/response.js

```diff
@@ -202,20 +202,11 @@ res.send = function send(body) {
  */
 
 res.json = function json(obj) {
-  var val = obj;
-
-  // support res.json(status, obj)
-  if (arguments.length === 2) {
-    deprecate('res.json(status, obj): Use res.status(status).json(obj) instead');
-    this.statusCode = arguments[0];
-    val = arguments[1];
-  }
-
   // settings
   var app = this.app;
   var replacer = app.get('json replacer');
   var spaces = app.get('json spaces');
-  var body = JSON.stringify(val, replacer, spaces);
+  var body = JSON.stringify(obj, replacer, spaces);
```

Comment:
nit: make the variable `val`, since we accept any value; it was only `obj` in the arguments so we could use `val` throughout the body without being hit by an argument reassignment deopt.